src/NuGet.Clients/PackageManagement.VisualStudio/Utility/VsHierarchyUtility.cs

```diff
@@ -37,6 +37,28 @@ public static IVsHierarchy ToVsHierarchy(Project project)
             return hierarchy;
         }
 
+        public static VsHierarchyItem GetHierarchyItemForProject(Project project)
+        {
+            Debug.Assert(ThreadHelper.CheckAccess());
+
+            return new VsHierarchyItem(ToVsHierarchy(project));
+        }
+
+        public static string GetProjectId(Project project)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            var hierarchyItem = GetHierarchyItemForProject(project);
+
+            Guid id;
+            if (!hierarchyItem.TryGetProjectId(out id))
+            {
+                id = Guid.Empty;
```

Comment:
Returning `null` (the absence of a value) or `string.Empty` makes more sense to me here than `{00000000-0000-0000-0000-000000000000}`.